### ISSUE TITLE
feat(ai-service): Train domain-specific intent classification model

### ISSUE THREAD

**Issue by Peleke** [Peleke/MindMirror#187]
## Problem

The current heuristic-based intent classification has ordering issues that cause misclassification:

| Input | Expected | Actual |
|-------|----------|--------|
| "how do I improve my squat form?" | ASK_COACH | CREATE_WORKOUT |
| "ate eggs and toast for breakfast" | LOG_MEAL | Empty/unclear |

**Root cause**: Workout keyword patterns are checked BEFORE question patterns, so "squat form" hits "squat" → CREATE_WORKOUT before the "how do I" question pattern is evaluated.

## Current State

- **Heuristic classifier** in `ai_orchestrator_graph.py` uses pattern matching with hardcoded order
- **LLM fallback** exists but is a TODO stub (`_classify_with_llm`)
- Confidence threshold of 0.6 never triggers LLM because heuristics return 0.75-0.95

## Quick Fix (In Progress)

Use LLM for intent classification with a well-crafted prompt. This gives us:
- Better accuracy for ambiguous cases
- Ability to understand question vs. command intent
- No hardcoded pattern ordering issues

## Long-term Solution

Train a **domain-specific small language model** for intent classification:

### Why a Custom Model?
1. **Latency**: LLM calls add 200-500ms per classification
2. **Cost**: Every input hits the LLM API
3. **Reliability**: No dependency on external API availability
4. **Privacy**: User inputs stay local

### Training Approach
1. **Collect labeled data** from production (with user consent)
2. **Generate synthetic data** using LLM for edge cases
3. **Train a small model** (DistilBERT, TinyBERT, or similar)
4. **Intents to classify**:
   - `CREATE_WORKOUT` - logging a workout session
   - `LOG_MEAL` - logging food/meals
   - `ASK_COACH` - questions about fitness, form, advice
   - `ANALYZE` - requesting data analysis
   - `CREATE_PROGRAM` - multi-week program creation
   - `QUICK_LOG` - water, sleep, weight, steps
   - `CLARIFY` - ambiguous input

### Example Training Data
```json
{"text": "5x5 squats at 315", "intent": "CREATE_WORKOUT"}
{"text": "how do I improve my squat form?", "intent": "ASK_COACH"}
{"text": "ate eggs and toast for breakfast", "intent": "LOG_MEAL"}
{"text": "why am I so sore after deadlifts?", "intent": "ASK_COACH"}
{"text": "bench press 3x10 185lbs", "intent": "CREATE_WORKOUT"}
{"text": "what's the best way to build muscle?", "intent": "ASK_COACH"}
```

## Acceptance Criteria

- [ ] LLM-based classification working as interim solution
- [ ] Collect 1000+ labeled examples from production
- [ ] Train small classification model (< 100MB)
- [ ] Achieve > 95% accuracy on held-out test set
- [ ] Deploy model with < 50ms inference latency
- [ ] A/B test against LLM classification
- [ ] Remove LLM dependency once model is validated

## Related

- PR #186 - Mega AI Plane with E2E testing
- E2E tests that validate intent classification
